internal/cluster/certificates.go

```diff
@@ -194,6 +195,17 @@ func (c Certificates) Lookup(ctx context.Context, ctrlclient client.Client, clus
 			return err
 		}
 		certificate.KeyPair = kp
+
+		// Ensure Secret is owned by the Cluster. This provides a migration path for Secrets
+		// previously owned by KubeadmConfigs.
+		patchHelper, err := patch.NewHelper(s, ctrlclient)
+		if err != nil {
+			return err
+		}
+		s.OwnerReferences = certificate.AsSecret(cluster).OwnerReferences
```

Comment:
This should probably only override the OwnerReferences if the OwnerRef is currently a `KubeadmConfig`, otherwise we might take ownership of a Secret we shouldn't.

Reply:
ah right, forgot about BYOCerts